### ISSUE TITLE
Potential accuracy issue native_batch_norm decomp

### ISSUE THREAD

**Issue by anijain2305** [pytorch/torchdynamo#1896]
### 🐛 Describe the bug

The following two cmds fail accuracy

`python benchmarks/dynamo/timm_models.py  --accuracy --device cuda --backend=aot_inductor_debug --training --float32 --only=resnest101e`

`python benchmarks/dynamo/torchbench.py --training --accuracy --device cuda --backend=aot_inductor_debug --amp --only=mobilenet_v3_large`

And both of them pass if I disable the `aten.native_batch_norm` in torch/_inductor/decomposition.py

Note that the backend here is `aot_inductor_debug` which only uses the decomp and partitioner from inductor but does not use inductor codegen.

**Comment by anijain2305** [comment]
Not sure as of now, but could be related to this - https://github.com/pytorch/pytorch/issues/88375

**Comment by eellison** [comment]
I will determine if this is related to input mutations.

**Comment by eellison** [comment]
Mobilenet is fixed by https://github.com/pytorch/pytorch/issues/88375 but resnet is not.

**Comment by eellison** [comment]
I'm not actively looking into the `resnest101e` error since i have a few other issues assigned, unassigning.

**Comment by anijain2305** [comment]
cc @janeyx99

**Comment by anijain2305** [comment]
Two other models that show similar symptoms are (added in the first comment as well)

`python benchmarks/dynamo/torchbench.py --training --accuracy --device cuda --backend=aot_inductor_debug --float32 --only=resnet50_quantized_qat`

`python benchmarks/dynamo/torchbench.py --training --accuracy --device cuda --backend=aot_inductor_debug --float32 --only=mobilenet_v2_quantized_qat`

**Comment by bdhirsh** [comment]
hmmm `resnet50_quantized_qat` is only failing with inductor - it runs fine when I run it with aot_eager. I'm running on top of this temporary stack for debugging, with a fixed batch norm + input mutation working.

**Comment by bdhirsh** [comment]
Oh, I see in the aot eager trace that we're calling `cudnn_batch_norm`. Maybe I can force the repro on eager if I get it to run native_batch_norm instead.

**Comment by bdhirsh** [comment]
Still confirming, but, I think I'm able to repro on aot eager, only if I:

(1) ensure that batchnorm decomposes to native_batch_norm
(2) enable the `native_batch_norm_legit_functional` decomposition inside of aot autograd. If I don't enable the decomp (and leave it as a primitive in the trace), the model passes.

That suggests a problem with the decomp

**Comment by eellison** [comment]
@bdhirsh can we close this ?

**Comment by bdhirsh** [comment]
I think the fact that the decomp degrades accuracy a bit is still something that we should try to look into. But given that Mark’s E2E accuracy tests now pass using the decomp, this probably isn’t hi-pri anymore
